### PR TITLE
for cds^4.4 (to be released)

### PR DESCRIPTION
**Pull request by sjvans**

### CODE REVIEW

**Comment by gregorwolf** [comment]
Hi @sjvans,

as CDS 4.4.4 is already available I've gave it a try. But it seems that your change doesn't work completely yet. Your original correction fails with:

`TypeError: Cannot read property 'query' of undefined`

as this.dbc isn't initialized. Changing it to0:

```JavaScript
    this.on(['BEGIN', 'COMMIT', 'ROLLBACK'], async function (req) {
      if (!this.dbc && req.event === 'BEGIN') {
        this.dbc = await this.acquire(req)
      }
      return this.dbc.query(req.event)
    })
```

doesn't solve it as the connection seems not to be released. But with our original code we currently get ["Release called on client which has already been released to the pool."](https://answers.sap.com/questions/13205540/sap-postgresql-hyperscaler-option-issue-in-reading.html).

Best regards
Gregor

File: package.json [hunk 1]
```diff
@@ -63,7 +63,7 @@
     "supertest": "^6.0.1",
     "uuidv4": "^6.2.4"
   },
-  "husky": {
+  "_husky": {
```
Comment (gregorwolf) [inline package.json:66]: Please do not simply deactivate husky. Please check out [CONTRIBUTING.md#other-thingies](https://github.com/sapmentors/cds-pg/blob/main/docs/CONTRIBUTING.md#other-thingies) how to format commit messages to be OK for the check.

Comment (sjvans) [reply package.json:66]: > CONTRIBUTING.md#other-thingies

hi @gregorwolf 

the pr was only meant as pointer, not as contribution ;)

tests are running now (cf. require of deploy), but an expect is not fulfilled anymore...

**Comment by gregorwolf** [comment]
OK, I've created #77 which includes this changes.

**Comment by Sandeep-Malhotra** [comment]
Hello
Kindly let me know as I am getting below mentioned error. 
[Release called on client which has already been released to the pool.](https://answers.sap.com/questions/13205540/sap-postgresql-hyperscaler-option-issue-in-reading.html
) 
I  have even tried both cds version 4.3.1 and 4.4.4 got the same issue
Thanks in advance.
Regards
Sandeep

**Comment by vobu** [comment]
> [Release called on client which has already been released to the pool.](https://answers.sap.com/questions/13205540/sap-postgresql-hyperscaler-option-issue-in-reading.html)
> I have even tried both cds version 4.3.1 and 4.4.4 got the same issue

please try the latest release.

**Comment by Sandeep-Malhotra** [comment]
All fixed thanks for providing a quick fix.
Really appreciate your assistance

**Comment by sjvans** [comment]
> But with our original code we currently get ["Release called on client which has already been released to the pool."](https://answers.sap.com/questions/13205540/sap-postgresql-hyperscaler-option-issue-in-reading.html).

hi @gregorwolf 

this looks like a compatibility issue between cds and cds-runtime. i.e., if you do an npm install now with `cds@4.3`, you will get `cds-runtime^2.7`, which are not compatible. hence, it might be best to add cds-runtime in the list of dependencies, e.g. `cds~4.4` and `cds-runtime~2.7`. additionally, every consuming project should have a `package-lock.json` to avoid this.

best,
sebastian

**Pull request closed without merging**